.github/workflows/ci.yml

```diff
@@ -0,0 +1,128 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - '*'
+      - '!*backport*'
+    tags:
+      - 'v*'
+      - '!*dev*'
+      - '!*pre*'
+      - '!*post*'
+  pull_request:
+  # Allow manual runs through the web UI
+  workflow_dispatch:
+
+jobs:
+  core:
+    uses: OpenAstronomy/github-actions-workflows/.github/workflows/tox.yml@main
+    with:
+      submodules: false
+      coverage: codecov
+      toxdeps: tox-pypi-filter
+      posargs: -n auto
+      libraries: |
+        apt:
+          - libopenjp2-7
+      envs: |
+        - linux: py39
+
+  test:
+    needs: [core]
+    uses: OpenAstronomy/github-actions-workflows/.github/workflows/tox.yml@main
+    with:
+      submodules: false
+      coverage: codecov
+      toxdeps: tox-pypi-filter
+      posargs: -n auto
+      libraries: |
+        apt:
+          - libopenjp2-7
+        brew:
+          - openjpeg
+      envs: |
+        - macos: py38
+        - windows: py310
+        - linux: py38-oldestdeps
+        - linux: py39-conda
+          libraries: ''
+
+  online:
+    needs: [core]
+    uses: OpenAstronomy/github-actions-workflows/.github/workflows/tox.yml@main
+    with:
+      default_python: '3.8'
+      submodules: false
+      coverage: codecov
+      toxdeps: tox-pypi-filter
+      posargs: -n auto --dist loadgroup
+      libraries: |
+        apt:
+          - libopenjp2-7
+      envs: |
+        - linux: build_docs_ci
+          posargs: ''
+          pytest: false
+          libraries:
+            apt:
+              - libopenjp2-7
+              - graphviz
+        - linux: py38-online
+
+  cron:
+    if: github.event_name == 'schedule' || github.event_name == 'workflow_dispatch'
+    uses: OpenAstronomy/github-actions-workflows/.github/workflows/tox.yml@main
+    with:
+      default_python: '3.8'
+      submodules: false
+      coverage: codecov
+      toxdeps: tox-pypi-filter
+      posargs: -n auto
+      libraries: |
+        apt:
+          - libopenjp2-7
+      envs: |
+        - linux: base_deps
+        - linux: py39-devdeps
+
+  publish:
```

Comment:
Rather than putting publish here, I think it would be better to put it in it's own workflow, which is triggered upon completion of the CI one, like this: https://github.com/DKISTDC/dkist/blob/main/.github/workflows/publish.yml#L3-L8

This will mean that it doesn't show up as a skipped check in all PRs.

Reply:
@Cadair Yeah, it would be good to remove all the skipped messages. It seems that in the DKIST repo the publish workflow will run even if the CI workflow has failures. [It can be configured to skip if failures like this](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#running-a-workflow-based-on-the-conclusion-of-another-workflow), although sunpy would need a way to ignore the status of the online tests when publishing which doesn't seem straightforward. However, [`Note: This event will only trigger a workflow run if the workflow file is on the default branch.`](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#workflow_run) sounds like this might not work for publishing on the 3.0 and 3.1 branches?